context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { useEventCallback } from '@mui/material/utils';
+import Menu from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import AddRounded from '@mui/icons-material/AddRounded';
+
+import SelectableTableProvider from 'js/shared-styles/tables/SelectableTableProvider';
+import { WorkspacesIcon } from 'js/shared-styles/icons';
+
+import { useOpenDialog } from '../../workspaces/WorkspacesDropdownMenu/WorkspacesDropdownMenu';
+import { useCreateWorkspaceForm } from '../../workspaces/NewWorkspaceDialog/useCreateWorkspaceForm';
+import NewWorkspaceDialog from '../../workspaces/NewWorkspaceDialog/NewWorkspaceDialog';
+import AddDatasetsFromSearchDialog from '../../workspaces/AddDatasetsFromSearchDialog/AddDatasetsFromSearchDialog';
+import { useAppContext, useFlaskDataContext } from '../../Contexts';
+import { useTrackEntityPageEvent } from '../useTrackEntityPageEvent';
+
+interface ProcessedDataWorkspaceMenuProps {
+  button: React.ReactNode;
+}
+
+function ProcessedDataWorkspaceMenu({ button }: ProcessedDataWorkspaceMenuProps) {
+  const {
+    entity: { mapped_data_access_level, hubmap_id, uuid, entity_type, status },
+  } = useFlaskDataContext();
+  const isDataset = entity_type === 'Dataset';
+  const { isWorkspacesUser } = useAppContext();
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+  const track = useTrackEntityPageEvent();
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    track({
+      action: 'Open Workspace Menu',
+      label: hubmap_id,
+    });
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const {
+    control,
+    errors,
+    removeDatasets,
+    setDialogIsOpen: setOpenCreateWorkspace,
+    dialogIsOpen: createWorkspaceIsOpen,
+    ...rest
+  } = useCreateWorkspaceForm({
+    defaultName: hubmap_id,
+    initialSelectedDatasets: [uuid] ?? [],
+  });
+
+  const openEditWorkspaceDialog = useOpenDialog('ADD_DATASETS_FROM_SEARCH');
+
+  const trackCreateWorkspace = useEventCallback(() => {
+    track({
+      action: 'Start Creating Workspace',
+      label: hubmap_id,
+    });
+    setOpenCreateWorkspace(true);
+    handleClose();
+  });
```

Comment:
Including 'track' in the name suggests its only purpose is tracking. I'd remove track from the fn name here and below.